### PR TITLE
Trading fixes and improvements

### PR DESCRIPTION
**Pull request by validgem**
Trading will no longer break if 
1. the villager can trade multiple items for the same input item (1.13+ if the trade isn't the first of such trades) 
2. the price is affected by demand or villager gossip in 1.14+

trading test is rewritten to test for all of that.

### CODE REVIEW

**Comment by rom1504** [comment]
please rebase on master

**Comment by validgem** [comment]
done !

File: lib/plugins/villager.js [hunk 1]
```diff
@@ -100,11 +99,15 @@ function inject (bot, { version }) {
...
-        inputItem2: Item.fromNotch(trade.inputItem2 || { blockId: -1 }),
-        outputItem: Item.fromNotch(trade.outputItem || { blockId: -1 })
-      }))
+      villager.trades = packet.trades.map(trade => {
+        const ret = Object.assign(trade, {
+          inputItem1: Item.fromNotch(trade.inputItem1 || { blockId: -1 }),
+          inputItem2: Item.fromNotch(trade.inputItem2 || { blockId: -1 }),
+          outputItem: Item.fromNotch(trade.outputItem || { blockId: -1 })
+        })
+        ret.realPrice = trade.demand !== undefined ? Math.min(Math.max((trade.inputItem1.count + trade.specialPrice + Math.max(0, Math.floor(trade.inputItem1.count * trade.demand * trade.priceMultiplier))), 1), trade.inputItem1.stackSize) : trade.inputItem1.count
```
Comment (extremeheat) [inline lib/plugins/villager.js:108]: can you split this to multiple lines?

Comment (validgem) [reply lib/plugins/villager.js]: of course , I made it cleaner now

**Review by extremeheat — APPROVED** [review]

**Review by validgem — COMMENTED** [review]
this is a bit unrelated , but the command for running specific tests in readme doesn't work

**Review by Karang — APPROVED** [review]
LGTM

**Pull request merged**